tests/util/test_createPREMISEventXML.py

```diff
@@ -0,0 +1,447 @@
+
+
+def test_has_eventIdentifier_element():
+    """
+    Check that the eventIdentifier tree is present.
+    """
+    premis = util.createPREMISEventXML(None, None, None, None)
+    eventIdentifier = premis.xpath(
+        '/p:event/p:eventIdentifier',
+        namespaces={'p': util.PREMIS_NAMESPACE}
+    )
+    assert len(eventIdentifier) == 1
+    assert len(eventIdentifier[0].getchildren()) == 2
+
+
+def test_eventIdentifierValue_element_text():
+    """
+    Verify the eventIdentifier kwarg yields an eventIdentifier element with
+    the text from the argument.
+    """
+    eventIdentifier = 'ABC'
+    premis = util.createPREMISEventXML(None, None, None, None,
+                                       eventIdentifier=eventIdentifier)
+    eventIdentifierValue = premis.xpath(
+        '/p:event/p:eventIdentifier/p:eventIdentifierValue',
+        namespaces={'p': util.PREMIS_NAMESPACE}
+    )
+    assert len(eventIdentifierValue) == 1
+    assert eventIdentifierValue[0].text == eventIdentifier
+
+
+def test_eventIdentifierValue_element_default_text():
+    """
+    Verify the eventIdentifier kwarg yields an eventIdentifier element with
+    the default text when it is falsy.
+    """
+    premis = util.createPREMISEventXML(None, None, None, None)
+    eventIdentifierValue = premis.xpath(
+        '/p:event/p:eventIdentifier/p:eventIdentifierValue',
+        namespaces={'p': util.PREMIS_NAMESPACE}
+    )
+    assert len(eventIdentifierValue) == 1
+    assert eventIdentifierValue[0].text is not None
+
+
+def test_eventIdentifierType_element_has_text():
+    """
+    Verify the eventIdentifierType always has text.
+    """
+    premis = util.createPREMISEventXML(None, None, None, None)
+
+    eventIdentifierType = premis.xpath(
+        '/p:event/p:eventIdentifier/p:eventIdentifierType',
+        namespaces={'p': util.PREMIS_NAMESPACE}
+    )
+    assert len(eventIdentifierType) == 1
+    assert eventIdentifierType[0].text is not None
+
+
+def test_eventDateTime_has_default_date():
+    """
+    Verify the eventDateTime has a datetime when the eventDate kwarg
+    is not set.
+    """
+    premis = util.createPREMISEventXML(None, None, None, None)
+
+    eventDateTime = premis.xpath(
+        '/p:event/p:eventDateTime',
+        namespaces={'p': util.PREMIS_NAMESPACE}
+    )
+    assert len(eventDateTime) == 1
+    assert eventDateTime[0].text is not None
+
+
+def test_eventDateTime_element_has_custom_datetime():
+    """
+    Verify the eventDate kwarg yields an eventDateTime element with
+    the text from the argument.
+    """
+    date = datetime(2015, 01, 01)
+    premis = util.createPREMISEventXML(None, None, None, None,
+                                       eventDate=date)
+    eventDateTime = premis.xpath(
+        '/p:event/p:eventDateTime',
+        namespaces={'p': util.PREMIS_NAMESPACE}
+    )
+    assert len(eventDateTime) == 1
+    assert eventDateTime[0].text == str(date)
+
+
+def test_eventDetails_element_text_is_none():
```

Comment:
Do you want this and the next function to be be named with _eventDetail_ (not eventDetails)